[locator] show field where text has been found in current layer filter

3nids opened the pull request:
also fix an issue that a field was not returning a match for big number (e/f formats)

3nids committed:
[locator] show field where text has been found

also fix an issue that a field was not returning a match for big number (e/f formats)

3nids commented on src/app/locator/qgsinbuiltlocatorfilters.cpp:
```diff
@@ -234,7 +234,7 @@ void QgsActiveLayerFeaturesLocatorFilter::prepare( const QString &string, const
     else if ( allowNumeric && field.isNumeric() )
     {
-      expressionParts << QStringLiteral( "%1 = %2" ).arg( QgsExpression::quotedColumnRef( field.name() ) ).arg( numericValue );
+      expressionParts << QStringLiteral( "%1 = %2" ).arg( QgsExpression::quotedColumnRef( field.name() ) ).arg( numericValue, 0, trunc( numericValue ) == numericValue ? 'f' : 'g' );
```
not sure this is the best approach? shall we use `f` in any case?

nyalldawson replied:
I've no strong opinions either way. Go with your gut 👍

nyalldawson commented:
Looks good to me!

3nids committed:
init layer tree model when testing QgisApp

3nids committed:
add test for inbuilt active layer locator filter

3nids committed:
use string comparison for numerical value

this is how it is done in attribute table filter and seems to have better result than comparing numerical values

3nids commented:
now using string comparison instead of numerical. Seems to work better, and the same technique is actually used in attribute table filter.

3nids committed:
switch back to numerical precision and cleanup layers

3nids commented:
@nyalldawson  still good for you?

3nids commented:
(please squash if you merge)
